### PR TITLE
CMSIS-NN: Add SVDF with 8 bit state tensor

### PR DESCRIPTION
**Pull request by hmogensen-arm**
Add CMSIS-NN support for SVDF kernel with int8 state tensor.  

Currently CMSIS-NN has support for SVDF with integer 16 bit state tensor. With this commit it will also have support for integer 8 bit state tensors.

#1087 

BUG=#1087

### CODE REVIEW

**Comment by hmogensen-arm** [comment]
label comp:micro:arm

**Comment by freddan80** [comment]
LGTM

File: tensorflow/lite/micro/kernels/cmsis_nn/svdf.cc [hunk 1]
```diff
@@ -265,208 +100,86 @@ void EvalIntegerSVDF(TfLiteContext* context, TfLiteNode* node,
...
+                         "Could not find matching function for type %s.",
+                         TfLiteTypeGetName(weights_time_tensor->type));
+      return kTfLiteError;
   }
-  micro_context->DeallocateTempTfLiteTensor(activation_state);
-  micro_context->DeallocateTempTfLiteTensor(output);
-  return kTfLiteOk;
 }
 
 TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
```
Comment (deqiangc) [inline tensorflow/lite/micro/kernels/cmsis_nn/svdf.cc:149]: to be consistent with name PrepareSvdf, prefer to rename this to EvalSvdf

File: tensorflow/lite/micro/kernels/cmsis_nn/svdf.cc [hunk 2]
```diff
@@ -265,208 +100,86 @@ void EvalIntegerSVDF(TfLiteContext* context, TfLiteNode* node,
...
 
-    case kTfLiteInt8: {
-      EvalIntegerSVDF(context, node, input, weights_feature, weights_time, bias,
-                      params, activation_state, output, data);
+    case kTfLiteInt8:
+    case kTfLiteInt16: {
+      return EvalIntegerSVDF(context, node, input, weights_feature,
+                             weights_time, bias, params, activation_state,
+                             output, data);
       return kTfLiteOk;
```
Comment (deqiangc) [inline tensorflow/lite/micro/kernels/cmsis_nn/svdf.cc:182]: I think this additional return should be removed

**Review by deqiangc — APPROVED** [review]

**Pull request merged**